MangaList/src/main/kotlin/model/Manga.kt

```diff
@@ -0,0 +1,23 @@
+package model
+
+internal data class Manga(val name: String) {
```

Comment:
- Không cần thiết phải dùng internal nhiều quá thế này, em để public cx đc
- với data class thì nên để tất thuộc tính lên đầu
- Access Mod: ưu tiên đặt là val. Nếu muốn thay đổi một thuộc tính thì '.copy()'. Sử dụng var thì biến của em có thể bị thay đổi ở bất cứ đâu. 
- Phần tag: với bài toán hiện tại thì ổn, nma trong tương lai em sẽ cần tách hẳn ra một model/controller mới, tên có thể là BookTag. Lí do thì ở Android, data muốn chuyển qua lại giữa các View, thì phải là một 'Parcelable' và Parcelable thì chỉ nhận 1) kiểu nguyên thuỷ 2)String. Các kiểu khác thì phải thêm một bước chuyển qua lại JSON

Reply:
Với các thuộc tính như url, chapter, rating thì chắc chắn sẽ thay đổi 1 lúc nào đó nên em để là `var`

Ngoài ra thì anh có thể giải thích vì sao sử dụng `.copy()` để khởi tạo 1 object mới sau đó mới thay đổi giá trị thuộc tính lại tốt hơn không ạ?

Reply:
câu hỏi đầu tiên, anh sẽ chia các class Model, hay gọi là data class ra làm 2 loại: 1 là plain data object, 2 là state holder. Ae phân biệt được hai cái này chứ? 
<details><summary>Spoil</summary>
<p>

- Plain data object: ví dụ như em query từ CSDL ra, một cái dữ liệu thời tiết Hà Nội hôm nay, một cái xe ô tô. Tên gọi khác của nó là POJO

- State holder: ví dụ như một AddressForm(name, dob, gender, age...)
</p>
</details> 

**Tại sao không nên dùng var?** 
Kotlin là Java, nên các object là kiểu dữ liệu tham chiếu (mn chú ý đọc về phần này). Trừ khi các em gọi Constructor, hoặc đối với data class thì có thêm `.copy`, hoặc một số techniques khác anh chưa muốn đề cập. Thì tất cả n cái object ở nhiều file, các em cứ tưởng nó là n cái khác nhau, nhưng không nó đều cùng là 1 cái instance. Em bắt đầu thấy vấn đề chưa, khi var là một kiểu dữ liệu tham chiếu. Một khi em đổi value cho nó, thì tất cả các chỗ sử dụng tới cái instance đó, sẽ được cập nhật theo. Thế nếu anh lỡ set cho cái var đấy thành null ở FileA.kt, nhưng FileB.kt anh cứ tu tu var.member là crash luôn app đúng ko?

**Tại sao sử dụng `.copy()`, và tại sao tốt nhất thì nên là val + copy** 
- A muốn code của anh [immutable nhất có thể](https://dev.to/rapasoft/on-immutability-in-kotlin-251k), đồng thời, một POJO mà cần update, thì tốt nhất nó nên là một object mới.
- Một lí do khác: ở Android java (không có data class, muốn compare by value phải override Comparable (nhớ khoá trước ko)) thế nên là gì, ko phải class nào cũng có compare by value được, nên nta xác định là compare by reference (tham chiếu luôn). Vậy nên để đảm bảo dữ liệu của anh luôn refresh nhất thì mình nên tạo hẳn một instance mới, chứ ko chỉnh sửa cái cũ.

**Data class var khi nào** 
A chưa từng dùng

Reply:
Về câu 3: var theo a chỉ nên sử dụng trong các class là stateholder